tensorboard/lazy.py

```diff
@@ -44,10 +44,14 @@ def wrapper(load_fn):
     # make future lookups efficient (only failed lookups call __getattr__).
     @_memoize
     def load_once(self):
+      if load_once.loading:
+        raise ImportError("Circular import when resolving LazyModule %r" % name)
```

Comment:
Optional but seems like you might as well put this logic in `_memoize` itself.

Reply:
But `_memoize` takes a function that accepts a single hashable argument,
so recursion is perfectly fine. If `_memoize` took a niladic function,
the argument would be more convincing, I think.

Reply:
If you don’t mind, I’ll keep it as written in this PR, because that
enables us to give a better error message.